### PR TITLE
[IMP] new fields and model for manage server notifications message for e...

### PR DESCRIPTION
**Pull request by archetipo**
...ach pec recipient

[IMP] new method in mail.message to check if main sent message is in error state
error is ture if not all recipients are in the notification list ,
or if there is at least one notification with error
[FIX] view remove old field add new field , add new pec partner tree view

### CODE REVIEW

File: l10n_it_pec_messages/model/mail_message.py [hunk 1]
```diff
@@ -26,6 +26,40 @@
 from openerp.osv import fields, orm
 
 
+class PecNoficaitons(orm.Model):
```
Comment (tafaRU) [inline l10n_it_pec_messages/model/mail_message.py]: It seems a typo, isn'it?

File: l10n_it_pec_messages/model/mail_message.py [hunk 2]
```diff
@@ -115,6 +165,49 @@ def _get_out_server(self, cr, uid, ids, name,
         'direction': 'in'
     }
 
+    def  ChckStatus(self, cr, uid, ids, context=None):
```
Comment (tafaRU) [inline l10n_it_pec_messages/model/mail_message.py]: Another typo? ;-)

File: l10n_it_pec_messages/model/mail_message.py [hunk 3]
```diff
@@ -26,6 +26,40 @@
...
+        ),
+        'recipient_addr': fields.related(
+            'name', 'recipient_id', 'pec_mail',
+            type='char',
+            string="Recipient Address", readonly=True,
+        ),
+        'type': fields.related(
+            'name', 'pec_type',
+            type='char',
+            string="Nitice Type", readonly=True,
```
Comment (tafaRU) [inline l10n_it_pec_messages/model/mail_message.py]: What Nitice stand for? Typo again? ;-)

File: l10n_it_pec_messages/model/mail_mail.py [hunk 4]
```diff
@@ -77,3 +78,34 @@ def send_get_email_dict(self, cr, uid, mail, partner=None, context=None):
...
+                        address_lst = mail.email_to
+                    partner_obj = self.pool.get('res.partner')
+                    existing_recipient_ids = partner_obj.exists(
+                        cr, SUPERUSER_ID, recipient_ids, context=context)
+                    for partner in partner_obj.browse(
+                        cr, SUPERUSER_ID, existing_recipient_ids,
+                        context=context
+                    ):
+                        address_lst += formataddr(
+                            (partner.name, partner.pec_mail)) + ' '
```
Comment (eLBati) [inline l10n_it_pec_messages/model/mail_mail.py]: Shouldn't addresses be separated by `', '`?

File: l10n_it_pec_messages/model/mail_message.py [hunk 5]
```diff
@@ -115,6 +168,59 @@ def _get_out_server(self, cr, uid, ids, name,
...
+    def CheckStatus(self, cr, uid, ids, context=None):
+        notif_pool = self.pool['pec.notifications']
+        if context is None:
+            context = {}
+        if not hasattr(ids, '__iter__'):
+            ids = [ids]
+        error_lst = []
+        noerror_lst = ids
+        if ids:
+            for row in self.browse(cr, uid, ids, context=context):
```
Comment (eLBati) [inline l10n_it_pec_messages/model/mail_message.py]: Please replace `row` with `message`

File: l10n_it_pec_messages/model/mail_mail.py (hunk 4 above)
Comment (archetipo) [inline l10n_it_pec_messages/model/mail_mail.py]: no i've tried but not work and this separation is made later here
https://github.com/OCA/OCB/blob/8.0/openerp/addons/base/ir/ir_mail_server.py#L323

Comment (eLBati) [inline l10n_it_pec_messages/model/mail_mail.py]: See https://github.com/archetipo/l10n-italy/pull/20

Comment (eLBati) [inline l10n_it_pec_messages/model/mail_mail.py]: https://github.com/archetipo/l10n-italy/pull/21

**Pull request merged**